music/src/transcription/model.ts

```diff
@@ -207,114 +196,41 @@ export class OnsetsAndFrames {
-        bwKernel, bwRecurrentKernel,
-        reorderGates(
-            getVar(`${scope}/bidirectional_rnn/bw/lstm_cell/bias`), 1.0)
-      ];
-    }
-
-    function denseWeights(scope: string) {
-      return [getVar(`${scope}/weights`), getVar(`${scope}/biases`)];
-    }
-
     tf.tidy(() => {
       // Onsets model has a conv net, followed by an LSTM. We separate the two
       // parts so that we can process the convolution in batch and then
       // flatten before passing through the LSTM.
-      const onsetsCnn = this.getAcousticCnn();
-      onsetsCnn.setWeights(convWeights('onsets'));
-      this.onsetsCnn = onsetsCnn;
-
-      const onsetsRnn = tf.sequential();
-      onsetsRnn.add(getBidiLstm([null, onsetsCnn.outputShape[2] as number]));
-      onsetsRnn.add(tf.layers.dense(
-          {units: MIDI_PITCHES, activation: 'sigmoid', trainable: false}));
-      onsetsRnn.setWeights(
-          lstmWeights('onsets').concat(denseWeights('onsets/onset_probs')));
-      this.onsetsRnn = onsetsRnn;
-
-      const activationCnn = this.getAcousticCnn();
-      activationCnn.add(tf.layers.dense(
-          {units: MIDI_PITCHES, activation: 'sigmoid', trainable: false}));
-      activationCnn.setWeights(
-          convWeights('frame').concat(denseWeights('frame/activation_probs')));
-      this.activationCnn = activationCnn;
+      this.onsetsCnn = new AcousticCnn();
+      this.onsetsCnn.setWeights(vars, 'onsets');
+      this.onsetsRnn = new BidiLstm([null, this.onsetsCnn.outputShape[2]]);
+      this.onsetsRnn.setWeights(vars, 'onsets', 'onset_probs');
+
+      this.activationCnn = new AcousticCnn('sigmoid');
+      this.activationCnn.setWeights(vars, 'frame', 'activation_probs');
 
       // Frame RNN takes concatenated ouputs of onsetsRnn and activationCnn
       // as its inputs.
-      const frameRnn = tf.sequential();
-      frameRnn.add(getBidiLstm([null, MIDI_PITCHES * 2]));
-      frameRnn.add(tf.layers.dense(
-          {units: MIDI_PITCHES, activation: 'sigmoid', trainable: false}));
-      frameRnn.setWeights(
-          lstmWeights('frame').concat(denseWeights('frame/frame_probs')));
-      this.frameRnn = frameRnn;
-
-      const velocityCnn = this.getAcousticCnn();
-      velocityCnn.add(tf.layers.dense(
-          {units: MIDI_PITCHES, activation: 'linear', trainable: false}));
-      velocityCnn.setWeights(
-          convWeights('velocity')
-              .concat(denseWeights('velocity/onset_velocities')));
-      this.velocityCnn = velocityCnn;
+      this.frameRnn = new BidiLstm([null, MIDI_PITCHES * 2]);
+      this.frameRnn.setWeights(vars, 'frame', 'frame_probs');
+
+      this.velocityCnn = new AcousticCnn('linear');
+      this.velocityCnn.setWeights(vars, 'velocity', 'onset_velocities');
     });
   }
 
-  /**
-   * Returns an acoustic stack without setting variables.
-   */
-  private getAcousticCnn() {
-    const acousticCnn = tf.sequential();
+  private logMessage(msg: string) {
```

Comment:
If you’re adding a stand-alone function, then think you should have some sort of flag to turn off debugging and wrap all the logs maybe in a if (verbose). I don’t think a lot of people like a super verbose library

Reply:
It's not really that verbose...also can't they filter based on .info?

Reply:
Should we have some sort of shared logging library if we're going to do stuff like that? Is there a standard one we should be using?

Reply:
#159

Reply:
Apparently console.info === console.log. But people can filter us completely if they want. 
If we were to have some setting, would we just have a global variable mm.verbose === true/false?

Reply:
See #160.